newIDE/app/src/EventsSheet/index.js

```diff
@@ -805,24 +810,32 @@ export class EventsSheetComponentWithoutHandle extends React.Component<
       inlineEditingAnchorEl: parameterContext.domEvent
         ? parameterContext.domEvent.currentTarget
         : null,
-      inlineEditingChangesMade: false,
       inlineEditingPreviousValue: instruction.getParameter(parameterIndex),
     });
   };
 
   closeParameterEditor = (shouldCancel: boolean) => {
-    if (shouldCancel) {
-      const { instruction, parameterIndex } = this.state.editedParameter;
-      if (!instruction || !this.state.inlineEditingPreviousValue) return;
-
+    const { instruction, parameterIndex } = this.state.editedParameter;
+    if (!instruction) {
+      return;
```

Comment:
Exceptionally, can we avoid a early return an do a:

```
if (instruction) {
  if (shouldCancel && ...)
  if (!shouldCancel && ...)
}

// and then the rest:
const { inlineEditingAnchorEl } = this.state;
this.setState(...)
```

I'm a bit paranoid but I want to be sure that if, for some reason, instruction is null, we still close the editor.